eclipse-che/che-operator
# feat: Gh20635 native auth kube

### What does this PR do?
Make it possible to deploy on kubernetes with native authentication. You still need to explicitly patch CheCluster with `nativeUserMode: true`.

I would also welcome any testing on both kubernetes and openshift, with and without devworkspaces.


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20635


### How to test this PR?


```bash
cat EOF << > /tmp/patch.yaml
spec:
  auth:
    nativeUserMode: true
EOF

chectl server:deploy \
     --installer operator \
     --platform minikube \
     --che-operator-image quay.io/mvala/che-operator:gh20635-nativeAuthKube \
     --che-operator-cr-patch-yaml /tmp/patch.yaml \
     --workspace-engine=dev-workspace
```

1. deploy ^
2. import dex certificate into the browser
3. open Che URL
4. login with `che@eclipse.org:admin` or `userN@che:password` where N = [1-5]
5. Che will use user identity to talk with kubernetes. You can check `RoleBindings` created in user's namespace. You can test by coping for example request to get devworkspaces from browser network traffic. It should not work when you remove the `RoleBinding`.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
